sdk/storage/azblob/container/client.go

```diff
@@ -160,6 +160,23 @@ func (c *Client) Delete(ctx context.Context, options *DeleteOptions) (DeleteResp
 	return resp, err
 }
 
+// Restore operation restore the contents and properties of a soft deleted container to a specified container.
+// For more information, see https://docs.microsoft.com/en-us/rest/api/storageservices/restore-container.
+func (c *Client) Restore(ctx context.Context, options *RestoreOptions) (RestoreResponse, error) {
```

Comment:
Container Name and version are required parameters to restore a container. Both of these should be in the func signature. Options can be an empty struct for now. Else the using function with default value will always fail. 

Should we check if this API is needed at container level or just at service level?

Reply:
This is the `container` client, we already have the container name (see the following lines where we obtain it).

Reply:
@jhendrixMSFT, should we add the deleted container version as a parameter of the function (this parameter is mandatory for restoring a container). We will also have the options as a parameter which will be empty for now. What do you think?
@nakulkar-msft, this is what you are referring to, right?

Reply:
If version is required for the API to work, then yeah it should be a required parameter and the options type empty.

Reply:
Added deleted container version as a parameter and also keeping empty options struct